### PR TITLE
fix #278977 Allow stripes to be highlighted in piano roll editor

### PR DESCRIPTION
**Pull request by blackears**
Can now control-click on keyboard area to toggle highlights on background bars.
Fixing spelling error.

### CODE REVIEW

File: mscore/pianokeyboard.cpp [hunk 1]
```diff
@@ -23,6 +23,8 @@
 #include "libmscore/part.h"
 #include "libmscore/drumset.h"
 
+#include <QGuiApplication>
```
Comment (Howard-C) [inline mscore/pianokeyboard.cpp]: If in `all.h`, remove this

File: mscore/preferences.cpp [hunk 2]
```diff
@@ -190,6 +190,7 @@ void Preferences::init(bool storeInMemoryOnly)
             {PREF_UI_PIANOROLL_DARK_NOTE_UNSEL_COLOR,              new ColorPreference(QColor("#1dcca0"))},
             {PREF_UI_PIANOROLL_DARK_NOTE_SEL_COLOR,                new ColorPreference(QColor("#ffff00"))},
             {PREF_UI_PIANOROLL_DARK_BG_BASE_COLOR,                 new ColorPreference(QColor("#3a3a3a"))},
+            {PREF_UI_PIANOROLL_DARK_BG_KEY_HILIGHT_COLOR,          new ColorPreference(QColor("#555577"))},
```
Comment (Howard-C) [inline mscore/preferences.cpp]: Maybe spell the full word `HIGHLIGHT`?

File: mscore/pianoview.h [hunk 3]
```diff
@@ -103,6 +103,7 @@ class PianoView : public QGraphicsView {
       bool inProgressUndoEvent;
       
       QList<PianoItem*> noteList;
+      char pitchHighlight[128];
```
Comment (Howard-C) [inline mscore/pianoview.h]: Not sure `char` is the right choice here

File: mscore/preferences.h [hunk 4]
```diff
@@ -73,6 +73,139 @@ enum class MusicxmlExportBreaks : char {
...
+#define PREF_UI_PIANOROLL_LIGHT_NOTE_UNSEL_COLOR            "ui/pianoroll/light/note/unselected/color"
+#define PREF_UI_PIANOROLL_LIGHT_NOTE_SEL_COLOR              "ui/pianoroll/light/note/selected/color"
+#define PREF_UI_PIANOROLL_LIGHT_BG_BASE_COLOR               "ui/pianoroll/light/background/base/color"
+#define PREF_UI_PIANOROLL_LIGHT_BG_KEY_HILIGHT_COLOR        "ui/pianoroll/light/background/keys/highlight/color"
+#define PREF_UI_PIANOROLL_LIGHT_BG_KEY_WHITE_COLOR          "ui/pianoroll/light/background/keys/white/color"
+#define PREF_UI_PIANOROLL_LIGHT_BG_KEY_BLACK_COLOR          "ui/pianoroll/light/background/keys/black/color"
+#define PREF_UI_PIANOROLL_LIGHT_BG_GRIDLINE_COLOR           "ui/pianoroll/light/background/gridLine/color"
+#define PREF_UI_PIANOROLL_LIGHT_BG_TEXT_COLOR               "ui/pianoroll/light/background/text/color"
+
+
```
Comment (Howard-C) [inline mscore/preferences.h]: These are already in `global` module, add your new preference keys in there, not here.

File: mscore/pianoview.h (hunk 3 above)
Comment (blackears) [reply mscore/pianoview.h]: I was trying to keep the memory footprint small.  What datatype would be better?

Comment (Howard-C) [reply mscore/pianoview.h]: Don't know, don't really see this type often... Anyway if `char` is indeed better, please specify `signed` or `unsigned`, see https://musescore.org/en/handbook/developers-handbook/finding-your-way-around/musescore-coding-rules#chars

File: global/settings/types/preferencekeys.h [hunk 5]
```diff
@@ -149,6 +149,7 @@
 #define PREF_UI_PIANOROLL_DARK_NOTE_UNSEL_COLOR             "ui/pianoroll/dark/note/unselected/color"
 #define PREF_UI_PIANOROLL_DARK_NOTE_SEL_COLOR               "ui/pianoroll/dark/note/selected/color"
 #define PREF_UI_PIANOROLL_DARK_BG_BASE_COLOR                "ui/pianoroll/dark/background/base/color"
+#define PREF_UI_PIANOROLL_DARK_BG_KEY_HIGHLIGHT_COLOR       "ui/pianoroll/dark/background/keys/hilight/color"
```
Comment (Spire42) [inline global/settings/types/preferencekeys.h:152]: Please rename this preference to `ui/pianoroll/dark/background/keys/highlight/color` (correcting the spelling of “highlight”).

**Pull request merged**